common/functions/src/scalars/strings/elt.rs

```diff
@@ -56,18 +57,39 @@ impl Function for EltFunction {
         Ok(true)
     }
 
-    fn return_type(&self, _args: &[DataType]) -> Result<DataType> {
+    fn return_type(&self, args: &[DataType]) -> Result<DataType> {
+        if !args[0].is_numeric() && args[0] != DataType::String && args[0] != DataType::Null {
+            return Err(ErrorCode::IllegalDataType(format!(
+                "Expected string or null, but got {}",
+                args[0]
+            )));
+        }
+        for arg in &args[1..] {
+            if !arg.is_numeric() && *arg != DataType::String && *arg != DataType::Null {
+                return Err(ErrorCode::IllegalDataType(format!(
+                    "Expected string or null, but got {}",
+                    arg
+                )));
+            }
+        }
         Ok(DataType::String)
     }
 
     fn eval(&self, columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
         let n_column = columns[0].column().cast_with_type(&DataType::Int64)?;
+
+        let columns = columns[1..]
+            .iter()
+            .map(|c| c.column())
+            .map(|c| c.cast_with_type(&DataType::String).unwrap())
```

Comment:
`.unwrap` --> `?`

.collect::<Result<Vec<DataColumn>>>();

Reply:
my assumption was it is safe to do unwrap here because args are checked to be able to be casted, anyway remove `unwrap`